logrotate.c

```diff
@@ -2554,6 +2554,11 @@ int main(int argc, const char **argv)
     logSetLevel(MESS_NORMAL);
     setlocale (LC_ALL, "");
 
+    /* Creates the dir for state file and moves old
+       state file to new location */
+    mkdir(STATEDIR, 0755);
+    rename(OLDSTATEFILE, STATEFILE);
+
```

Comment:
We are using /var/lib for Linux.
(Linux)
#ifndef STATEFILE
#define STATEFILE "/var/lib/logrotate.status"
#endif

#ifdef __hpux
#define STATEFILE "/var/run/logrotate.status"
#endif